modules/Microsoft.ContainerService/managedClusters/main.bicep

```
@@ -619,6 +631,20 @@ module managedCluster_roleAssignments '.bicep/nested_roleAssignments.bicep' = [f
   }
 }]
 
+resource dnsZone 'Microsoft.Network/dnsZones@2018-05-01' existing = if (dnsZoneResourceId != null && webApplicationRoutingEnabled) {
+  name: last(split(dnsZoneResourceId, '/'))!
```

I guess there is a chance that the DNS Zone is in a different subscription / resource Group? In this case I'd suggest as smal update:

```suggestion
  name: last(split(dnsZoneResourceId, '/'))!
  scope: resourceGroup(split(dnsZoneResourceId, '/')[2], split(dnsZoneResourceId, '/')[4])
```